core/serialization_v6.c

```diff
@@ -954,15 +954,15 @@ uint32_t chunk_v6_read_shape(Stream *s,
             case P3S_CHUNK_ID_SHAPE_NAME: {
                 uint8_t nameLen;
                 memcpy(&nameLen, cursor, sizeof(uint8_t));
+                cursor = (void *)((uint8_t *)cursor + 1);
                 name = malloc(nameLen + 1);
                 if (name == NULL) {
                     cclog_error("malloc failed");
-                    // TODO: handle error
-                    // (if `nameStr` is NULL, we cannot let the function continue)
+                } else {
+                    memcpy(name, cursor, sizeof(char) * nameLen);
+                    name[nameLen] = 0;
                 }
-                cursor = (void *)((uint8_t *)cursor + 1);
-                memcpy(name, cursor, sizeof(char) * nameLen);
-                name[nameLen] = 0;
+                cursor = (void *)((uint8_t *)cursor + nameLen);
```

Comment:
this cursor pointer update was missing before and if the name chunk wasn't the last subchunk of the shape chunk, this lead to an error.